src/main/org/deidentifier/arx/masking/DataMaskingFunction.java

```diff
@@ -92,41 +92,65 @@ private String getRandomAlphanumericString(char[] buffer, Random random) {
         }
     }
     
+    /**
+     * Generates a random permutation column's rows
+     * 
+     * @author giupardeb
+     * 
+     */
+    
     public static class PermutationFunctionColumns extends DataMaskingFunction {
     	
         /** SVUID */
-		private static final long serialVersionUID = 1470074649699937850L;
+    	private static final long serialVersionUID = 1470074649699937850L;
     	
-    	public PermutationFunctionColumns (boolean ignoreMissingData) {
-            super(ignoreMissingData, false);
-        }
+    	private final String typePermutation;
+		
+		/**
+         * Creates a new instance
+         * @param ignoreMissingData
+         */
+    	public PermutationFunctionColumns (boolean ignoreMissingData, String typePermutation) {
```

Comment:
About  "typePermutation": how many different types do you expect to implement? If not more than one, I suggest to remove this parameter. Otherwise, I suggest to use an enum, e.g. PermutationType.FYKY.

Reply:
I edit PermutationFunctionColumns class:

`    	private final PermutationType typePermutation;
    	
    	/**
    	 * create a class enum to define permutation type
    	 * 
    	 * @author giupardeb
    	 *
    	 */
    	public enum PermutationType {
    		FYKY
    	}
		
	/**
         * Creates a new instance
         * @param ignoreMissingData
         * 
         */
    	public PermutationFunctionColumns (boolean ignoreMissingData, PermutationType typePermutation) {
    		super(ignoreMissingData, false);
    		this.typePermutation = typePermutation;
    	}

		@Override
		public void apply(DataColumn column) {
			switch(typePermutation) {
			case FYKY:
				fisherYatesKnuthYao(column);
			break;
			}	
		}`